### PR TITLE
Improve Automated Test Coverage

### PR DESCRIPTION
**Pull request by shinigami-777**
This is part of [JENKINS-69756](https://issues.jenkins.io/browse/JENKINS-69756).
Added test for `NodeLabelBuildParameter` class  specifically covering the `MacroEvaluationException` case in the `getAction` method.

Test coverage before:
nblp1

Test coverage after:
nblp2

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -92,6 +92,18 @@
...
+      <artifactId>junit-jupiter</artifactId>
+      <version>5.10.0</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.mockito</groupId>
+      <artifactId>mockito-inline</artifactId>
+      <version>4.0.0</version>
+      <scope>test</scope>
+    </dependency>
```
Comment (MarkEWaite) [inline pom.xml:106]: I'm unwilling to add a dependency on the outdated mockito-inline library when it has been replaced with mockito-core.  If you must use mock objects, please use mockito-core.  Since there are no other tests in the plugin based on mock objects, I'd very much prefer that this test also not use mock objects.
```suggestion
```

File: src/main/java/org/jvnet/jenkins/plugins/nodelabelparameter/parameterizedtrigger/NodeLabelBuildParameter.java [hunk 2]
```diff
@@ -38,12 +38,12 @@ public Action getAction(AbstractBuild<?, ?> build, TaskListener listener) throws
             labelExpanded = TokenMacro.expandAll(build, listener, labelExpanded);
         } catch (MacroEvaluationException e) {
             labelExpanded = nodeLabel;
+            listener.getLogger().println("Token expansion failed.");
```
Comment (MarkEWaite) [inline src/main/java/org/jvnet/jenkins/plugins/nodelabelparameter/parameterizedtrigger/NodeLabelBuildParameter.java:41]: The code is already printing a full stack trace in case of macro expansion failure.  How does the extra text help the user?
```suggestion
```

File: src/main/java/org/jvnet/jenkins/plugins/nodelabelparameter/parameterizedtrigger/NodeLabelBuildParameter.java [hunk 3]
```diff
@@ -38,12 +38,12 @@ public Action getAction(AbstractBuild<?, ?> build, TaskListener listener) throws
...
             e.printStackTrace(listener.getLogger());
         }
         LabelParameterValue parameterValue =
                 new LabelParameterValue(name, labelExpanded, false, new AllNodeEligibility());
         listener.getLogger().println("define: " + parameterValue);
-
```
Comment (MarkEWaite) [inline src/main/java/org/jvnet/jenkins/plugins/nodelabelparameter/parameterizedtrigger/NodeLabelBuildParameter.java:46]: This seems like an unnecessary formatting change.  Not harmful, but

File: src/test/java/org/jvnet/jenkins/plugins/nodelabelparameter/parameterizedtrigger/NodeLabelBuildParameterTest.java [hunk 4]
```diff
@@ -23,6 +23,11 @@
  */
 package org.jvnet.jenkins.plugins.nodelabelparameter.parameterizedtrigger;
 
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (MarkEWaite) [inline src/test/java/org/jvnet/jenkins/plugins/nodelabelparameter/parameterizedtrigger/NodeLabelBuildParameterTest.java:26]: The rest of the plugin uses JUnit 4 assertions.  Please stay with JUnit 4 assertions in this file as well
```suggestion
```

File: pom.xml [hunk 5]
```diff
@@ -92,6 +92,18 @@
       <artifactId>throttle-concurrents</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.junit.jupiter</groupId>
+      <artifactId>junit-jupiter</artifactId>
+      <version>5.10.0</version>
+      <scope>test</scope>
+    </dependency>
```
Comment (MarkEWaite) [inline pom.xml:100]: The plugin is already using JUnit 4.  Please stay with JUnit 4.  If JUnit 5 is required, please use the most recent JUnit 5 release instead of an older release.
```suggestion
```

**Review by MarkEWaite — CHANGES_REQUESTED** [review]
Thanks for the effort to increase test coverage, but there is so much use of mock objects that it reduces the value of the test.  Based on the [token-macro source code](https://github.com/search?q=repo%3Ajenkinsci%2Ftoken-macro-plugin%20MacroEvaluationException&type=code), there should not be a need to use mock objects for the test.

**Comment by shinigami-777** [comment]
> Thanks for the effort to increase test coverage, but there is so much use of mock objects that it reduces the value of the test. Based on the [token-macro source code](https://github.com/search?q=repo%3Ajenkinsci%2Ftoken-macro-plugin%20MacroEvaluationException&type=code), there should not be a need to use mock objects for the test.

Yes I looked at the TokenMacro source code and found the string to cause the exception without mocking. Thanks a lot for suggesting.

File: pom.xml (hunk 5 above)
Comment (shinigami-777) [reply pom.xml]: Changed it back to use JUnit 4.

File: pom.xml (hunk 1 above)
Comment (shinigami-777) [reply pom.xml]: Removed the use of mock objects entirely.

Comment (MarkEWaite) [reply pom.xml]: That's great!  Thanks very much for adapting to my comments, especially when they were very general.

File: pom.xml (hunk 5 above)
Comment (MarkEWaite) [reply pom.xml]: Thanks!

**Review by MarkEWaite — APPROVED** [review]
Thanks very much!  Your test implementation looks very good.  I adapted it with some additions that I found while exploring the test from inside a debugger.  Specific details of each change are embedded in the commit comments for each of the commits:

* https://github.com/jenkinsci/nodelabelparameter-plugin/pull/366/commits/49c5d389109de7b5de52f65214a0a77c4682d257 - simplify the node label expression
* https://github.com/jenkinsci/nodelabelparameter-plugin/pull/366/commits/0ab1adac0a14ae3c3af276d071d953602d2e198e - use a non-deprecated StreamTaskListener constructor
* https://github.com/jenkinsci/nodelabelparameter-plugin/pull/366/commits/3ac804e4864941dec91516388ea2d6ab9bd3e4f0 - run the build and assert that it passes
* https://github.com/jenkinsci/nodelabelparameter-plugin/pull/366/commits/e4797920ab8c5fa0aae6f5cf1bb05d903698ad02 - use hamcrest matchers for better failure messages

Hopefully you're OK with my changes.  I plan to merge this as soon as the CI checks pass.
